sklearn/metrics/base.py

```diff
@@ -131,3 +131,64 @@ def _average_binary_score(binary_metric, y_true, y_score, average,
         return np.average(score, weights=average_weight)
     else:
         return score
+
+
+def _average_multiclass_ovo_score(binary_metric, y_true, y_score, average):
+    """Uses the binary metric for one-vs-one multiclass classification,
+    where the score is computed according to the Hand & Till (2001) algorithm.
+
+    Parameters
+    ----------
+    y_true : array, shape = [n_samples]
+        True multiclass labels.
+        Currently only handles labels with values 0 to n_classes - 1.
+
+    y_score : array, shape = [n_samples, n_classes]
+        Target scores corresponding to probability estimates of a sample
+        belonging to a particular class
+
+    average : string, ['macro' (default), 'weighted']
+        ``'macro'``:
+            Calculate metrics for each label, and find their unweighted
+            mean. This does not take label imbalance into account. Classes
+            are assumed to be uniformly distributed.
+        ``'weighted'``:
+            Calculate metrics for each label, taking into account the a priori
+            distribution of the classes.
+
+    binary_metric : callable, the binary metric function to use.
+        Accepts the following as input
+            y_true' : array, shape = [n_samples']
```

Comment:
The intent was to differentiate the input to `binary_metric` from the top-level `y_true` and `n_samples` because the `binary_metric` function might only be called on a sub-array of `y_true`. Should I remove the `'` and just replace with something like `y_true_target` and `y_score_target`?